src/NuGet.Core/NuGet.Build.Tasks/NuGet.targets

```diff
@@ -179,7 +179,8 @@ Copyright (c) .NET Foundation. All rights reserved.
       Interactive="$(NuGetInteractive)"
       RestoreForceEvaluate="$(RestoreForceEvaluate)"
       RestorePackagesConfig="$(RestorePackagesConfig)"
-      EmbedFilesInBinlog="$(RestoreEmbedFilesInBinlog)">
+      EmbedFilesInBinlog="$(RestoreEmbedFilesInBinlog)"
+      SDKAnalysisLevel="$(SDKAnalysisLevel)">
```

Comment:
For my learning, does the SDKAnalysisLevel property require an older version of the SDK to be installed along with the newer version?

Reply:
It does not. It can be used in one SDK version. Assume a user installed a newer SDK version but does not yet have a way to deal with the new warnings and errors that come with the new SDK. That is when they can use SDKAnalysisLevel to tell their new SDK they don't want these new warnings and errors by lowering the analysis level. As a result, whenever we introduce new errors and warnings, we should make sure they are considerate of the SDKAnalysisLevel

Reply:
I don't fully understand why NuGet should have the SDKAnalysisLevel property ready for future consumption. Given that this is a new feature, I think we need a spec to fully understand the motivation behind this change. I have a few questions, such as why we need these properties in nuget.exe and what future requirements we have that necessitate this property. I was under the impression that we could use the new SDK analysis property similar to Andy's changes in the [commit](https://github.com/NuGet/NuGet.Client/commit/073a073d13b319f2d1d96af29d77f9dacd759053).